tests/Aspire.Hosting.Azure.Tests/AzureManifestUtils.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Nodes;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Utils;
+
+public sealed class AzureManifestUtils
+{
+    public static async Task<(JsonNode ManifestNode, string BicepText)> GetManifestWithBicep(IResource resource, bool skipPreparer = false)
+    {
+        if (!skipPreparer)
+        {
+            var executionContext = new DistributedApplicationExecutionContext(DistributedApplicationOperation.Publish);
+            var azurePreparer = new AzureResourcePreparer(Options.Create(new AzureProvisioningOptions()), executionContext);
+            var resources = (IResourceCollection)Activator.CreateInstance(
+                typeof(IResourceCollection).Assembly.GetType("Aspire.Hosting.ApplicationModel.ResourceCollection")!)!;
+            resources.Add(resource);
+            await azurePreparer.BeforeStartAsync(new DistributedApplicationModel(resources), cancellationToken: default);
```

Comment:
What is this jank?

Reply:
Which part do you hate?
* Having to call AzurePreparer before generating the bicep?
* Reusing the internal ResourceCollection class via Reflection? If this is an issue, I can create a copied implementation of IResourceCollection.

Reply:
I've updated the code to no longer use reflection. I added a convienence ctor on DistributedApplicationModel. Let me know if you think this is acceptable.